### PR TITLE
CLN: address TODOs

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/lib.pyx [hunk 1]
```diff
@@ -221,7 +221,7 @@ def is_iterator(obj: object) -> bool:
     return PyIter_Check(obj)
 
 
-def item_from_zerodim(val: object) -> object:
+cpdef object item_from_zerodim(object val):
```
Comment (jreback) [inline pandas/_libs/lib.pyx:224]: can you add inline here?

File: pandas/_libs/lib.pyx [hunk 2]
```diff
@@ -2344,11 +2344,8 @@ def map_infer_mask(ndarray arr, object f, const uint8_t[:] mask, bint convert=Tr
         else:
             val = f(arr[i])
 
-            if cnp.PyArray_IsZeroDim(val):
-                # unbox 0-dim arrays, GH#690
-                # TODO: is there a faster way to unbox?
-                #   item_from_zerodim?
-                val = val.item()
+            # unbox 0-dim arrays, GH#690
+            val = item_from_zerodim(val)
```
Comment (jreback) [inline pandas/_libs/lib.pyx:2348]: does this auto call the cdef version?

**Review by jorisvandenbossche — APPROVED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/lib.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:2348]: yes

File: pandas/_libs/lib.pyx (hunk 1 above)
Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:224]: could i guess, i dont think it makes a difference

Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:224]: on balance i prefer not to mix "inline" with "cpdef"

Comment (jreback) [reply pandas/_libs/lib.pyx:224]: right that’s the question does this make a perf difference? as u have it vs master

Comment (jbrockmendel) [reply pandas/_libs/lib.pyx:224]: huh, if map_infer is somewhat slower in this PR than in master, and its _worse_ with `inline`.  so im going to revert this edit, remove the TODO comments, and in a separate branch look into if we can improve item_from_zerodim

Comment (jbrockmendel) [reply pandas/_libs/lib.pyx]: updated+green

Comment (WillAyd) [reply pandas/_libs/lib.pyx]: I can open an issue about this separately, but our current inline strategy _may_ be doing more harm than good. GCC has a `-Winline` flag that will tell you when a function declared with inline isn't actually inlined, and running this on Linux appears to produce 11 MB worth of warnings with a clean build of pandas

Comment (jbrockmendel) [reply pandas/_libs/lib.pyx]: Yikes.  Addressing the larger issue separately makes sense.  For the purposes of this PR, is there something that should be changed?

Comment (WillAyd) [reply pandas/_libs/lib.pyx]: Things lgtm here as is, but let's see what Jeff thinks

Comment (WillAyd) [reply pandas/_libs/lib.pyx]: raised issue in #33926

Comment (jreback) [reply pandas/_libs/lib.pyx]: no this is fine, but yeah looking at this generally is prob a good idea.

**Review by jreback — APPROVED** [review]

**Pull request merged**